README.md

```diff
@@ -21,233 +24,132 @@ repositories {
+            itemView.song_item.text = item.title
+            itemView.setOnClickListener { item.onSongClick(item.id) }
+            imageLoader.load(item.imageUrl).into(itemView.song_cover_iv)
         }
     }
 }
-```
-
-Setup recycler view:
-
-```java
-final UniversalAdapter adapter = new UniversalAdapter(Collections.<ViewHolderManager>singletonList(new DataViewHolder()));
-recyclerView.setAdapter(adapter);
-```
-
-Give new data to adapter:
-
-```java
-adapter.call(Arrays.toList(new Data(1, "Cow"), new Data(2, "Dg"), new Data(3, "Cat"));
-```
-
-And another data so recycler view will be nice animated:
-
-```java
-adapter.call(Arrays.toList(Data(2, "Dog"), new Data(3, "Cat"), new Data(4, "Elephant"));
-```
-
-### With auto-value (recommended)
 
-Usage like above with small improvement:
-
-```java
-@AutoValue
-private class Data implements BaseAdapterItem {
-
-    @Nonnull
-    @AdapterId
-    public abstract String id();
-    @Nonnull
-    public abstract String name();
-    public abstract int color();
-
-    public Data create(@Nonull String id, @Nonnull String name, int color) {
-        return AutoValue_Data(id, name, color);
+class FooterViewHolder : LayoutViewHolderManager<FooterItem>(
+    R.layout.item_footer, FooterItem::class, { ViewHolder(it) }
+) {
+    class ViewHolder(itemView: View) : ViewHolderManager.BaseViewHolder<FooterItem>(itemView) {
+        override fun bind(item: FooterItem) {}
     }
-
-    // methods equal, adapterId, matches and same will be generated for you
 }
-```
-
-
-For more info look: [AutoValue: BaseAdapterItem Extension](https://github.com/m-zagorski/auto-value-base-adapter-item)
 
+```
 
-### Without Universal Adapter
-
-Implement some Pojo with your data:
-
-```java
-private class Data implements SimpleDetector.Detectable<Data> {
-
-    private final long id;
-    @Nonnull
-    private final String name;
-    private final int color;
-
-    Data(long id, @Nonnull String name, int color) {
-        this.id = id;
-        this.name = name;
-        this.color = color;
-    }
+- Setup adapter and bind data:
 
-    /**
-     * Return true if id matches
-     */
-    @Override
-    public boolean matches(@Nonnull Data item) {
-        return ((Data) item).id == id;
-    }
+```kotlin
+val adapter = UniversalAdapter(listOf(headerViewHolder, songViewHolder, footerViewHolder))
+recyclerView.adapter = adapter
 
-    /**
-     * Return true if items are equal
-     */
-    @Override
-    public boolean same(@Nonnull Data item) {
-        return equals(item);
-    }
-
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (!(o instanceof Data)) return false;
-        final Data data = (Data) o;
-        return id == data.id &&
-                name.equals(data.name) &&
-                color == data.color;
-    }
-}
+// Whenever new data are applied, the RecyclerView is nicely animated
+viewModel.adapterItems.subscribe { adapter.submitList(it) }
```

Comment:
I think in the example RxJava can mislead users to think that it is required.

I'd say:
```kt
adapter.submitList([
  HeaderItem(test="Album 1"),
  Song(title="Album"),
  Song(title="Album"),
  HeaderItem(test="Album 1"),
  Song(title="Album"),
  Footer,
]);
```